src/ActionList/Item.tsx

```diff
@@ -59,9 +60,50 @@ export interface ItemProps extends React.ComponentPropsWithoutRef<'div'>, SxProp
    * Designates the group that an item belongs to.
    */
   groupId?: string
+
+  /**
+   * Items that are disabled can not be clicked, selected, or navigated through.
+   */
+  disabled?: boolean
+
+  /**
+   * Callback that will trigger both on click selection and keyboard selection.
+   */
+  onAction?: (item: Partial<ItemProps>, event?: MouseEvent | KeyboardEvent) => void
+}
+
+const getItemVariant = (variant = 'default', disabled?: boolean) => {
+  if (disabled) {
+    return {
+      color: themeGet('colors.text.disabled'),
+      iconColor: themeGet('colors.text.disabled'),
+      annotationColor: themeGet('colors.text.disabled'),
+      hoverBackground: 'inherit',
+      hoverCursor: 'default'
+    }
+  }
+
+  switch (variant) {
+    case 'danger':
+      return {
+        color: themeGet('colors.text.danger'),
+        iconColor: themeGet('colors.icon.danger'),
+        annotationColor: themeGet('colors.text.disabled'),
+        hoverBackground: themeGet('colors.bg.danger'),
+        hoverCursor: 'pointer'
+      }
+    default:
+      return {
+        color: 'inherit',
+        iconColor: themeGet('colors.text.disabled'),
+        annotationColor: themeGet('colors.text.disabled'),
+        hoverBackground: themeGet('colors.selectMenu.tapHighlight'),
+        hoverCursor: 'pointer'
+      }
+  }
 }
 
-const StyledItem = styled.div<{variant: ItemProps['variant']} & SxProp>`
+const StyledItem = styled.div<{variant: ItemProps['variant']; item?: ItemProps} & SxProp>`
```

Comment:
Can we remove `variant` here and just use `item`?

Reply:
I don't think we can here... I'm not 100% sure why, but `item` does not include `variant` here, `variant` is only a sibling key to `item`.

Reply:
Hm, I'm a bit confused by this since it looks like you're literally extracting the type for `variant` from `ItemProps`. Don't want to block progress, so approving, but I would like to understand this better.